### PR TITLE
Error page layout

### LINKED ISSUE #18098: dashboard-next: Show proper error page when failed to load keycloak settings due self-signed TLS
### Is your task related to a problem? Please describe.
Dashboard Next shows loader endless when there is self-signed TLS problem on keycloak host:
Screenshot_20201012_160647

It's needed to show proper error page when failed to load keycloak settings due self-signed TLS. Here is what the current dashboard displays:
Screenshot_20201012_160434

### PR DESCRIPTION
**Pull request by akurinnoy**
This PR introduces an error page layout and startup errors reporter service.

Demo:
Screenshot 2020-12-07 at 18 26 40

fixes https://github.com/eclipse/che/issues/18098

### CODE REVIEW

**Review by sleshchenko — APPROVED** [review]
The screenshot LGTM

File: src/components/Fallback/index.tsx [hunk 1]
```diff
@@ -13,8 +13,10 @@
 import { Spinner, spinnerSize } from '@patternfly/react-core';
 import React from 'react';
 
-export const fallback = (
+const Fallback = (
```
Comment (olexii4) [inline src/components/Fallback/index.tsx:16]: I am proposing to use in this case a function instead of a constant.

File: src/index.tsx [hunk 2]
```diff
@@ -30,7 +30,9 @@ const store = configureStore(history);
 
 const ROOT = document.querySelector('.ui-container');
 // preload app data
-new PreloadData(store).init().then(() => {
-  console.log('UD: preload data complete successfully.');
-  ReactDOM.render(<Provider store={store}><App history={history} /></Provider>, ROOT);
-});
+new PreloadData(store).init()
+  .then(() => console.log('UD: preload data complete successfully.'))
+  .catch(() => console.log('UD: preload data failed.'))
```
Comment (olexii4) [inline src/index.tsx]: In this case we are losing an error. What do you think about:
```
.catch(e => console.log(`UD: preload data failed. ${e}`))
```

**Review by olexii4 — APPROVED** [review]

File: src/components/Fallback/index.tsx (hunk 1 above)
Comment (akurinnoy) [reply src/components/Fallback/index.tsx:16]: We can do that later if we need to provide a property to this component.

**Pull request merged**